src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md

```diff
@@ -0,0 +1,322 @@
+
+```xml
+<?xml version="1.0"?>
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Event/etc/events.xsd">
+    <event name="checkout_onepage_controller_success_action">
+        <observer name="paymentfields_paymentfields_observer_frontend_sales_orderpaymentsavebefore" instance="Learning\CustomField\Observer\Frontend\Sales\OrderPaymentSaveBefore" />
+    </event>
+</config>
+```
+
+`app/code/Learning/CustomField/Observer/Frontend/Sales/OrderPaymentSaveBefore.php`
+
+{% collapsible Show code %}
+
+```php?start_inline=1
+<?php
+
+namespace Learning\CustomField\Observer\Frontend\Sales;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\OfflinePayments\Model\Purchaseorder;
+use Magento\Framework\App\Request\DataPersistorInterface;
+
+class OrderPaymentSaveBefore implements \Magento\Framework\Event\ObserverInterface
+{
+    protected $order;
+    protected $logger;
+    protected $_serialize;
+    protected $quoteRepository;
+
+    public function __construct(
+        \Magento\Sales\Api\Data\OrderInterface $order,
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Framework\Serialize\Serializer\Serialize $serialize
+    ) {
+        $this->order = $order;
+        $this->quoteRepository = $quoteRepository;
+        $this->logger = $logger;
+        $this->_serialize = $serialize;
+    }
+    /**
+     * Execute observer
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return void
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $orderids = $observer->getEvent()->getOrderIds();
+        if(!$orderids){
+            foreach ($orderids as $orderid) {
+                $order = $this->_order->load($orderid);
+                $method = $order->getPayment()->getMethod();
+                if($method == 'purchaseorder') {
+                    $quote_id = $order->getQuoteId();
+                    $quote = $this->quoteRepository->get($quote_id);
+                    $paymentQuote = $quote->getPayment();
+                    $paymentOrder = $order->getPayment();
+                    $paymentOrder->setData('paymentpocomment',$paymentQuote->getPaymentpocomment());
+                    $paymentOrder->save();
+                }
+            }
+        }
+    }
+}
+```
+
+{% endcollapsible %}
+
+## Step 6: Verify that the module works
+
+Add the product to the cart and go to the checkout page and select `purchase order` payment method and check that the new custom field can be seen (Purchase Order Comment).
+
+If the `purchase order` payment method cannot be seen on the checkout page then go to Magento Admin and enable the payment method. See **Stores** > **Settings** > **Configuration** > **Sales** > **Payment Methods** > **Other Payment Methods** > **Purchase Order**
```

Comment:
Making sure that purchase orders are enabled should be a prerequisite task. Move this info to near the beginning of this topic.

```suggestion
If the `purchase order` payment method cannot be seen on the checkout page, then go to Magento Admin and enable the payment method. See **Stores** > **Settings** > **Configuration** > **Sales** > **Payment Methods** > **Other Payment Methods** > **Purchase Order**.
```

Reply:
Moved this info to the beginning of the page.